staging/src/k8s.io/client-go/dynamic/client.go

```diff
@@ -220,9 +250,15 @@ func (rc *ResourceClient) Update(obj *unstructured.Unstructured) (*unstructured.
 	if len(obj.GetName()) == 0 {
 		return result, errors.New("object missing name")
 	}
+	resourceName, subresourceName := rc.parseResourceSubresourceName()
 	err := rc.cl.Put().
 		NamespaceIfScoped(rc.ns, rc.resource.Namespaced).
-		Resource(rc.resource.Name).
+		Resource(resourceName).
+		SubResource(subresourceName...).
```

Comment:
How about adding explicit methods like `UpdateStatus()`, etc instead of deriving the name? It would keep it backward compatible too.

I had added `UpdateStatus` and modified `Patch` to support subresources in https://github.com/kubernetes/kubernetes/pull/55168/commits/c961ac4b3e4c2f90c434eaa54b18acff90057d06#diff-781d9a2143af0fe2be85e527bf53626a.

Reply:
That's how we do in typed clients:

https://github.com/kubernetes/kubernetes/blob/ae0f45e855381d3731f8e66e536d1ad093ea9948/pkg/client/clientset_generated/internalclientset/typed/apps/internalversion/statefulset.go#L35-L47

Reply:
We also need to support `Update` for other subresources like `certificates.k8s.io/v1beta1/certificatesigningrequests/{name}/approval` and `v1/namespaces/{name}/finalize`.